### PR TITLE
Add Checkstyle rule to avoid statically importing copyOf and valueOf

### PR DESCRIPTION
**Pull request by wenleix**

### CODE REVIEW

File: src/checkstyle/checks.xml [hunk 1]
```diff
@@ -56,8 +56,8 @@
         <property name="message" value="Math.toIntExact should only be used with static imports" />
     </module>
     <module name="RegexpSingleline">
-        <property name="format" value="^import static .*\.of;$" />
-        <property name="message" value="Methods named &quot;of&quot; should never be statically imported" />
+        <property name="format" value="^import static .*\.(of|copyOf|valueOf);$" />
+        <property name="message" value="Methods named &quot;of&quot;, &quot;copyOf&quot; and &quot;valueOf&quot; should never be statically imported" />
```
Comment (electrum) [inline src/checkstyle/checks.xml]: This should technically be "or" instead of "and"

Comment (electrum) [reply src/checkstyle/checks.xml]: Or how about we change this to
```
The following methods may not be statically imported: of, copyof, valueOf

Comment (findepi) [reply src/checkstyle/checks.xml]: Or just `This method cannot be statically imported` (the error marker will tell everything that needs to be told).

**Comment by wenleix** [comment]
Thanks! I removed the quotes! I feel the method names still provide some information so I keep it :)

**Comment by haozhun** [comment]
I find the method names helpful here.

**Pull request merged**